### PR TITLE
Add a syntax style configuration file to the Dafny repository

### PR DESCRIPTION
**Pull request by keyboardDrummer**
Add a `.editorconfig` file that configures the Dafny syntax style. This file is supported by Rider, ReSharper, Visual Studio and dotnet-format.

### Outdated description

ReSharper and Rider allow storing settings in two layers, the "Personal" and the "Team Shared" layer, the personal layer is stored in `<SolutionName>sln.*DotSettings.user` and the team layer in `<SolutionName>sln.*DotSettings`. 

Several of the comments I've had on previous PRs were because my code wasn't following the Dafny style conventions. This PR adds a team shared settings file to the repository configured with some of Dafny's syntax style conventions.

Configurations:
- Always use braces in then/else blocks of if statements
- Let method opening brace start and the end of the line instead of on a new line.
- Don't automatically wrap lines

### CODE REVIEW

**Comment by camrein** [comment]
As I've had similar issues with the language server, it might be an idea to go for the [.editorconfig](https://editorconfig.org/) format since it is IDE agnostic. I've already used it in the [original repository](https://github.com/dafny-lang/language-server-csharp/blob/master/.editorconfig) before but removed it to not interfere with the compiler.

**Comment by keyboardDrummer** [comment]
> As I've had similar issues with the language server, it might be an idea to go for the [.editorconfig](https://editorconfig.org/) format since it is IDE agnostic. I've already used it in the [original repository](https://github.com/dafny-lang/language-server-csharp/blob/master/.editorconfig) before but removed it to not interfere with the compiler.

I'm honestly a bit confused by the concept of editorconfig. Since editorconfig is language agnostic, the amount of configuration it can offer for a specific language is so limited. You can see that the `dotnet format` tool which uses editorconfig has defined several times more custom properties than editorconfig has, [link](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-style-rule-options?view=vs-2019). These custom properties work for Microsoft C# tools, but not for JetBrains's, so the benefit of editorconfig, that it works for all tools, is lost.

For C# development, both JetBrain's and Microsoft tools are valid options, so I suggest we include a style configuration file for both. Keeping them in sync doesn't seem like a problem to me since they're small files that shouldn't change much over time. We should also run either Microsoft's `dotnet format` or JetBrain's `cleanupcode` as part of the build but that's out of scope for this PR.

**Comment by camrein** [comment]
> These custom properties work for Microsoft C# tools, but not for JetBrains's, so the benefit of editorconfig, that it works for all tools, is lost.
I don't use JetBrains myself, so I cannot really tell how well they integrate with editorconfig. According to [Rider's documentation](https://www.jetbrains.com/help/rider/Using_EditorConfig.html) it appeared to me that they at least supported a subset of the C# specific settings (since version 2021.1), so I thought it would be neat if the rules you configured could also be used for Visual Studio. According to the documentation's comments, the editorconfig-Editor of Rider may report keys as unknown even though the C# engine actually understands them. But I guess the support is still limited to an extent that there's no real benefit.

**Comment by keyboardDrummer** [comment]
> According to [Rider's documentation](https://www.jetbrains.com/help/rider/Using_EditorConfig.html) it appeared to me that they at least supported a subset of the C# specific settings (since version 2021.1)

O, sorry, I hadn't seen that and what I previously said was wrong then. I've tried to see if the custom properties from JetBrains overlap with MS's, and it seems there's a partial overlap. Some of the MS custom properties also occur in the giant list of JB ones.

Seems like we could try to configure everything in one `editorconfig` file, even if sometimes we have to specify the same thing with a customer JB and a custom MS property.

**Comment by keyboardDrummer** [comment]
Okay, it seems like all we need from .editorconfig is supported by both JB and MS, and that JB is trying to support all the properties that MS supports. Updating my version of Rider from a late 2020 to a 2021 version added support for properties that were missing.

**Comment by camrein** [comment]
Great, thanks a lot for taking a second look at it. If I could give some input, I'd recommend leaving the `root=true` away. This allows developers to configure additional settings outside of the repository. This is especially handy if you have installed many Roslyn Analyzers and want to use one global configuration for all your projects.

**Comment by keyboardDrummer** [comment]
> Great, thanks a lot for taking a second look at it. If I could give some input, I'd recommend leaving the `root=true` away. This allows developers to configure additional settings outside of the repository. This is especially handy if you have installed many Roslyn Analyzers and want to use one global configuration for all your projects.

Sounds good, and all the thanks should to you ^^

File: Source/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+end_of_line = lf
+
+#### .NET Coding Conventions ####
+
+#### C# Formatting Rules ####
+
+# New line preferences
+csharp_new_line_before_open_brace = namespace, types
+csharp_new_line_before_catch = false
+csharp_new_line_before_else = false
```
Comment (robin-aws) [inline Source/.editorconfig:23]:
```suggestion
csharp_new_line_before_else = false

```

**Review by robin-aws — APPROVED** [review]

**Review by robin-aws — APPROVED** [review]

**Pull request merged**